File: HLPProject/HLPProject/hlp/Interpreter.js
import { tryFind } from "fable-core/Map";
import { incPC, readPC } from "./State";
import { fsFormat } from "fable-core/String";
export function interpret(state, instr) {
  interpret: while (true) {
    var matchValue = tryFind(readPC(state), instr);
    var $var163 = matchValue != null ? matchValue.Case === "Instr" ? [0, matchValue.Fields[0]] : matchValue.Case === "Terminate" ? [1] : [2] : [2];

    switch ($var163[0]) {
      case 0:
        state = $var163[1](incPC(state));
        instr = instr;
        continue interpret;

      case 1:
        return state;

      case 2:
        return fsFormat("Undefined interpreting error")(function (x) {
          throw new Error(x);
        });
    }
  }
}

File: HLPProject/HLPProject/SourceCode_07March_Team1/Interpreter.fs
// The interpreter, this reads instructions and calls the appropriate function.

namespace Interpret
module Interpreter =
    
    open Common.State
    open Parse.Parser

    /// Calls functions on map of (memloc * instructions).
    let rec interpret state instr =
        match Map.tryFind (readPC state) instr with
        | Some(Instr(f)) -> interpret (f (incPC state)) instr
        | Some(Terminate) -> state
        | _ -> failwithf "Undefined interpreting error"